programs/drift/src/instructions/user.rs

```diff
@@ -653,6 +653,11 @@ pub fn handle_deposit<'c: 'info, 'info>(
     } else {
         DepositExplanation::None
     };
+    let signer = if ctx.accounts.authority.key() == user.authority {
```

Comment:
i think you want to flip this and log if it's a third party making the deposit?

```suggestion
    let signer = if ctx.accounts.authority.key() != user.authority {
```